### PR TITLE
[Enhancement](Load) Nereids supports http_stream and group_commit with stream load

### PR DESCRIPTION
**Pull request by zzzzzzzs**
## Proposed changes

This PR includes 2 functions:
- http_stream use nereids
- Backend of group_commit uses http_stream with nereids

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zzzzzzzs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 1]
```diff
@@ -410,12 +411,20 @@ public PlanFragment visitPhysicalOlapTableSink(PhysicalOlapTableSink<? extends P
             slotDesc.setIsNullable(column.isAllowNull());
             slotDesc.setAutoInc(column.isAutoInc());
         }
-        OlapTableSink sink = new OlapTableSink(
+        OlapTableSink sink;
+        // This statement is only used in the group_commit mode in the http_stream
+        if (context.getConnectContext().isGroupCommitStreamLoadSql()) {
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:416]: This is used for `insert into values` in Nereids, can remove it and add it in next PR.

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by zzzzzzzs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 2]
```diff
@@ -2921,6 +2926,92 @@ public List<Type> getReturnTypes() {
...
+                throw new AnalysisException("label and group_commit can't be set at the same time");
+            }
+            ((NativeInsertStmt) parsedStmt).isGroupCommitStreamLoadSql = true;
+        }
+        analyze(context.getSessionVariable().toThrift());
+    }
+
+    public void generateStreamLoadPlan(TUniqueId queryId) throws Exception {
+        try {
+            generateStreamLoadNereidsPlan(queryId);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:3000]: should use nereids planner only if session variable enable_nereids_planner=true and enable_nereids_dml=true. if we cannot get session level variable, we should use global level variable

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 3]
```diff
@@ -2921,6 +2926,92 @@ public List<Type> getReturnTypes() {
...
+        } catch (NereidsException | ParseException e) {
+            if (context.getMinidump() != null) {
+                MinidumpUtils.saveMinidumpString(context.getMinidump(), DebugUtil.printId(context.queryId()));
+            }
+            // try to fall back to legacy planner
+            LOG.debug("nereids cannot process statement\n" + originStmt.originStmt
+                    + "\n because of " + e.getMessage(), e);
+            LOG.debug("fall back to legacy planner on statement:\n{}", originStmt.originStmt);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:3008]: we should check enable_fallback_to_original_planner == true before fall back

Comment (zzzzzzzs) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java (hunk 2 above)
Comment (zzzzzzzs) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java]: done

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by zzzzzzzs** [comment]
run p0

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by zzzzzzzs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 4]
```diff
@@ -2967,6 +2972,141 @@ public List<Type> getReturnTypes() {
...
+        analyze(context.getSessionVariable().toThrift());
+        HttpStreamParams httpStreamParams = new HttpStreamParams();
+        httpStreamParams.setTxnId(insertStmt.getTransactionId());
+        httpStreamParams.setDb(insertStmt.getDbObj());
+        httpStreamParams.setTable(insertStmt.getTargetTable());
+        httpStreamParams.setLabel(insertStmt.getLabel());
+        return httpStreamParams;
+    }
+
+    public HttpStreamParams generateStreamLoadPlan(TUniqueId queryId) throws Exception {
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:3051]: It's better to name as `generateHttpStreamPlan`

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java [hunk 5]
```diff
@@ -117,8 +122,8 @@ private void runInternal(ConnectContext ctx, StmtExecutor executor) throws Excep
         // check auth
         if (!Env.getCurrentEnv().getAccessManager()
                 .checkTblPriv(ConnectContext.get(), targetTableIf.getDatabase().getCatalog().getName(),
-                        targetTableIf.getDatabase().getFullName(), targetTableIf.getName(),
-                        PrivPredicate.LOAD)) {
+                    targetTableIf.getDatabase().getFullName(), targetTableIf.getName(),
+                    PrivPredicate.LOAD)) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java:126]: Don't make unnecessary changes

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java [hunk 6]
```diff
@@ -150,14 +155,15 @@ private void runInternal(ConnectContext ctx, StmtExecutor executor) throws Excep
...
+                    throw new AnalysisException("group commit is not supported in Nereids now");
                 }
                 OlapTable olapTable = (OlapTable) targetTableIf;
                 insertExecutor = new OlapInsertExecutor(ctx, olapTable, label, planner, insertCtx);
                 boolean isEnableMemtableOnSinkNode =
                         olapTable.getTableProperty().getUseSchemaLightChange()
-                                ? insertExecutor.getCoordinator().getQueryOptions().isEnableMemtableOnSinkNode()
-                                : false;
+                            ? insertExecutor.getCoordinator().getQueryOptions().isEnableMemtableOnSinkNode()
+                            : false;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java:166]: Don't make unnecessary changes

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java (hunk 4 above)
Comment (zzzzzzzs) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java (hunk 5 above)
Comment (zzzzzzzs) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java (hunk 6 above)
Comment (zzzzzzzs) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java]: done

**Comment by zzzzzzzs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 7]
```diff
@@ -2969,6 +2974,141 @@ public List<Type> getReturnTypes() {
...
+            // Maybe our bug
+            LOG.debug("Command (" + originStmt.originStmt + ") process failed.", e);
+            context.getState().setError(ErrorCode.ERR_UNKNOWN_ERROR, e.getMessage());
+            throw new NereidsException("Command (" + originStmt.originStmt + ") process failed.",
+                    new AnalysisException(e.getMessage(), e));
+        }
+        return httpStreamParams;
+    }
+
+    private HttpStreamParams generateStreamLoadLegacyPlan(TUniqueId queryId) throws Exception {
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:3026]: name to `generateHttpStreamLegacyPlan`

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 8]
```diff
@@ -2969,6 +2974,141 @@ public List<Type> getReturnTypes() {
         return exprToType(parsedStmt.getResultExprs());
     }
 
+    private HttpStreamParams generateStreamLoadNereidsPlan(TUniqueId queryId) {
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:2977]: name to `generateHttpStreamNereidsPlan`

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 9]
```diff
@@ -415,12 +416,20 @@ public PlanFragment visitPhysicalOlapTableSink(PhysicalOlapTableSink<? extends P
...
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:421]: Here, can we set the session variable `group_commit` in `FrontendServiceImpl` and use here？
In this way, we does not need add a new variable in ConnectContext

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java [hunk 10]
```diff
@@ -118,9 +123,9 @@ private void runInternal(ConnectContext ctx, StmtExecutor executor) throws Excep
         TableIf targetTableIf = InsertUtils.getTargetTable(logicalQuery, ctx);
         // check auth
         if (!Env.getCurrentEnv().getAccessManager()
-                .checkTblPriv(ConnectContext.get(), targetTableIf.getDatabase().getCatalog().getName(),
-                        targetTableIf.getDatabase().getFullName(), targetTableIf.getName(),
-                        PrivPredicate.LOAD)) {
+            .checkTblPriv(ConnectContext.get(), targetTableIf.getDatabase().getCatalog().getName(),
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java:126]: remove unnecessary change

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java [hunk 11]
```diff
@@ -152,14 +157,15 @@ private void runInternal(ConnectContext ctx, StmtExecutor executor) throws Excep
...
+                    throw new AnalysisException("group commit is not supported in Nereids now");
                 }
                 OlapTable olapTable = (OlapTable) targetTableIf;
                 insertExecutor = new OlapInsertExecutor(ctx, olapTable, label, planner, insertCtx);
                 boolean isEnableMemtableOnSinkNode =
-                        olapTable.getTableProperty().getUseSchemaLightChange()
-                                ? insertExecutor.getCoordinator().getQueryOptions().isEnableMemtableOnSinkNode()
-                                : false;
+                    olapTable.getTableProperty().getUseSchemaLightChange()
+                        ? insertExecutor.getCoordinator().getQueryOptions().isEnableMemtableOnSinkNode()
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoTableCommand.java:167]: remove unnecessary change

**Review by mymeiyi — APPROVED** [review]

**Comment by mymeiyi** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]
